bit_matrix.go

```diff
@@ -0,0 +1,98 @@
+package rsmt2d
+
+import "fmt"
+
+type bitMatrix struct {
+	mask       []uint64
+	squareSize int
+}
+
+func newBitMatrix(squareSize int) bitMatrix {
+	bits := squareSize * squareSize
+	return bitMatrix{mask: make([]uint64, (bits+63)/64), squareSize: squareSize}
+}
+
+// i = rowIndex*squareSize+colIdx
+func (bm bitMatrix) SetFlat(i int) {
+	bm.mask[i/64] |= uint64(1) << uint(i%64)
+}
+
+func (bm bitMatrix) Get(row, col int) bool {
+	assertValidIndices(row, col, bm.squareSize)
+	idx := row*bm.squareSize + col
+	return bm.mask[idx/64]&(uint64(1)<<uint(idx%64)) > 0
+}
+
+func (bm *bitMatrix) Set(row, col int) {
+	assertValidIndices(row, col, bm.squareSize)
+	idx := row*bm.squareSize + col
+	bm.mask[idx/64] |= uint64(1) << uint(idx%64)
+}
+
+func (bm bitMatrix) ColumnIsOne(c int) bool {
+	for r := 0; r < bm.squareSize; r++ {
+		if !bm.Get(r, c) {
+			return false
+		}
+	}
+	return true
+}
+
+func (bm bitMatrix) RowIsOne(r int) bool {
+	for c := 0; c < bm.squareSize; c++ {
+		if !bm.Get(r, c) {
+			return false
+		}
+	}
+	return true
+}
+
+func (bm bitMatrix) NumOnesInRow(r int) int {
+	var counter int
+	for i := 0; i < bm.squareSize; i++ {
+		if bm.Get(r, i) {
```

Comment:
```suggestion
	for c := 0; c < bm.squareSize; c++ {
		if bm.Get(r, c) {
```

Reply:
addressed all these renamings in https://github.com/lazyledger/rsmt2d/pull/33/commits/3eca5806440382bbe899fe1d66700fedc9c6c2aa too